File: gui/modules/ranger/ranger.rb
name 'RANGER'
hp 6
foreground [60, 100, 15]
background [60, 100, 97]

left = 1.5 * MM_PER_HP
right = width - left
center = (left + right) / 2.0

y = 14.5
delta_y = 15.0

separator(y: y)

y += delta_y * 0.75

medium_knob(x: left, y: y, label: 'LEVEL')
output_port(x: right, y: y)

y += delta_y
cv_port(x: left, y: y)
attenuverter(x: right, y: y)

y += delta_y / 2.0
limit_top = y
separator(y: y)
y += delta_y * 0.75

medium_knob(x: left, y: y, label: 'CCW')
medium_knob(x: right, y: y, label: 'CW')

y += delta_y
attenuverter(x: left, y: y)
attenuverter(x: right, y: y)

y += delta_y
cv_port(x: left, y: y)
cv_port(x: right, y: y)

y += delta_y
polarity_toggle(x: left, y: y)
polarity_toggle(x: right, y: y)

y += delta_y * 0.6
limit_bottom = y
separator(y: limit_bottom)
line(x1:center, y1: limit_top, x2: center, y2: limit_bottom)

File: panelgen/internal/panel/ranger.go
package panel

import (
	"dhemery.com/DHE-Modules/panelgen/internal/svg"
)

func init() {
	registerBuilder("ranger", ranger)
}

func ranger() *Panel {
	const (
		hue = 60
		hp  = 6
	)
	var (
		fg = svg.HslColor(hue, 1, .15)
		bg = svg.HslColor(hue, 1, .97)
	)
	p := NewPanel("RANGER", hp, fg, bg, "ranger")

	const (
		width  = float64(hp) * mmPerHp
		left   = 1.5 * mmPerHp
		center = width / 2.0
		right  = width - left
	)

	y := 14.5
	dy := 15.0
	p.HSeparator(y)

	y += dy * 0.75
	p.MediumKnob(left, y, "LEVEL")
	p.OutPort(right, y, "OUT")

	y += dy
	p.CvPort(left, y)
	p.Attenuverter(right, y)

	y += dy / 2.0
	limitTop := y
	p.HSeparator(y)

	y += dy * 0.75
	p.MediumKnob(left, y, "CCW")
	p.MediumKnob(right, y, "CW")

	y += dy
	p.Attenuverter(left, y)
	p.Attenuverter(right, y)

	y += dy
	p.CvPort(left, y)
	p.CvPort(right, y)

	y += dy
	p.LevelRangeSwitch(left, y, 1)
	p.LevelRangeSwitch(right, y, 1)

	y += dy * 0.6
	limitBottom := y
	p.HSeparator(y)
	p.VLine(center, limitTop, limitBottom)

	return p
}
